addons/mail/models/mail_thread.py

```diff
@@ -4507,9 +4507,16 @@ def _message_update_content(self, message, body, attachment_ids=None, partner_id
         if strict:
             self._check_can_update_message_content(message.sudo())
 
-        msg_values = {
-            'body': escape(body),  # keep html if already Markup, otherwise escape
-        } if body is not None else {}
+        msg_values = (
+            {
+                "body": escape(body)  # keep html if already Markup, otherwise escape
+                + Markup(
+                    "<span class='o-mail-message-edited'/>"
+                ),
+            }
+            if body is not None
+            else {}
+        )
```

Comment:
Try to write simple statements :) when having one value in a dict, no need to enter complex writings.
```suggestion
        msg_values = {}
        if not is_html_empty(body):
            # keep html if already Markup, otherwise escape
            msg_values["body"] = escape(body) + Markup(
                "<span class='o-mail-message-edited'/>"
            )
```